### ISSUE TITLE
Use `pl.sink_*` in `LazyPolarsDataset._save`

### ISSUE THREAD

**Issue by AnH0ang** [kedro-org/kedro-plugins#702]
## Description
When passing a lazy DataFrame to `LazyPolarsDataset`, it is currently collected into an eager DataFrame before writing it using the appropriate `pl.write_*` function. This can be skipped by writing the lazy dataframe using `pl.scan_*`.

## Context
In some cases, it may be faster to collect the lazy DataFrame in streaming mode. Additionally, it is not always possible to collect the entire DataFrame (e.g., if the data is too large). Using `pl.sink_`, the entire data set does not need to be loaded.

## Possible Implementation
In the `_save` function, the input DataFrame could first be coerced into a lazy DataFrame and then written to disk using `pl.sink_*`.

**Comment by MatthiasRoels** [comment]
I would hold off such a change for now. As mentioned in the [docs](https://docs.pola.rs/api/python/stable/reference/api/polars.LazyFrame.sink_parquet.html):

> Streaming mode is considered unstable. It may be changed at any point without it being considered a breaking change.

On top of that, I heard the Polars team is working on completely rewriting their streaming engine. So I would just stick with the current implementation...

**Comment by astrojuanlu** [comment]
Indeed, streaming mode is unstable, but lazy non-streaming methods are considered stable, as far as I understand?

About what to do with remote storages, maybe we can offer a `sink_*` path for local files?

**Comment by MatthiasRoels** [comment]
Actually, the sink methods also use the streaming engine under the hood. Hence these methods should also be considered unstable as is explicitly mentioned in the [docs](https://docs.pola.rs/api/python/stable/reference/api/polars.LazyFrame.sink_parquet.html). So I wouldn’t recommend using it either.

**Comment by astrojuanlu** [comment]
New streaming engine is coming https://github.com/pola-rs/polars/issues/20947
